util.go

```diff
@@ -41,16 +41,19 @@ func SelectEqual(v string) func(string) bool {
 	}
 }
 
-func strToBytes(str string) []byte {
-	s := *(*reflect.StringHeader)(unsafe.Pointer(&str))
-	b := &reflect.SliceHeader{Data: s.Data, Len: s.Len, Cap: s.Len}
-	return *(*[]byte)(unsafe.Pointer(b))
+func strToBytes(str string) (bts []byte) {
```

Comment:
I would suggest the "safe version" of stringHeader and sliceHeader from reflect package, it may give more predictable memory safety.

```golang
// sliceHeader is a safe version of SliceHeader used within this package.
type sliceHeader struct {
	Data unsafe.Pointer
	Len  int
	Cap  int
}

// stringHeader is a safe version of StringHeader used within this package.
type stringHeader struct {
	Data unsafe.Pointer
	Len  int
}
```